### PR TITLE
Generate adoc copies in /content/en/docs to #78

### PR DESCRIPTION
**Pull request by rolfedh**
# Changes

I converted all the .md files in ./content/en/docs/ to .adoc files.

Convert shipwright doc files to asciidoc format.
We use these files (https://github.com/shipwright-io/build/tree/main/docs) 
to populate https://github.com/shipwright-io/website/tree/main/content/en/docs

Fixes https://github.com/shipwright-io/website/issues/75

# Submitter Checklist

- [y] Includes tests if functionality changed/was added
- [n] Includes docs if changes are user-facing
- [y] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [y] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by adambkaplan** [comment]
@rolfedh I rebased this on top of #80, and when I did so the converted docs don't show up :(.

**Comment by rolfedh** [comment]
> @rolfedh I rebased this on top of #80, and when I did so the converted docs don't show up :(.

@adambkaplan One of the error messages, `activating bundler-2failed` led me to [this blog post](https://jameshfisher.com/2019/01/09/how-to-fix-activating-bundler-2-failed-error-in-your-jekyll-build/). Although the post is about Jekyll, not Hugo, trimming the following two lines from the bottom of the `Gemfile.lock` file seems to have fixed it.
```
BUNDLED WITH
   2.3.7
```

File: content/en/docs/_index.adoc [hunk 1]
```diff
@@ -1,62 +1,59 @@
...
-    weight: 20
----
-
-
+= Documentation
+:draft: false
+:linkTitle: Documentation
+:menu: {"main"=>{"weight"=>20}}
+:no_list: true
+:weight: 20
```
Comment (adambkaplan) [inline content/en/docs/_index.adoc:6]: This is Hugo front matter - I believe Hugo expects this to exist in YAML format somewhere in the source content. Can you try reverting this to see what happens?

File: content/en/docs/api/build.adoc [hunk 2]
```diff
@@ -1,104 +1,129 @@
...
-    - [Defining the Source](#defining-the-source)
-    - [Defining the Strategy](#defining-the-strategy)
-    - [Defining ParamValues](#defining-paramvalues)
-    - [Defining the Builder or Dockerfile](#defining-the-builder-or-dockerfile)
-    - [Defining the Output](#defining-the-output)
-  - [BuildRun deletion](#BuildRun-deletion)
-
-## Overview
+= Build
+:weight: 10
```
Comment (adambkaplan) [inline content/en/docs/api/build.adoc:2]: Likewise - revert this so we get the Hugo front matter back.

File: content/en/docs/api/buildrun.adoc [hunk 3]
```diff
@@ -1,88 +1,87 @@
...
-  - [Understanding the state of a BuildRun](#understanding-the-state-of-a-buildrun)
-  - [Understanding failed BuildRuns](#understanding-failed-buildruns)
-  - [Step Results in BuildRun Status](#step-results-in-buildrun-status)
-  - [Build Snapshot](#build-snapshot)
-- [Relationship with Tekton Tasks](#relationship-with-tekton-tasks)
-
-## Overview
+= BuildRun
+:draft: false
+:weight: 40
```
Comment (adambkaplan) [inline content/en/docs/api/buildrun.adoc:3]: Revert for front matter.

File: content/en/docs/api/buildstrategies.adoc [hunk 4]
```diff
@@ -0,0 +1,696 @@
+= BuildStrategy and ClusterBuildStrategy
+:weight: 20
```
Comment (adambkaplan) [inline content/en/docs/api/buildstrategies.adoc:2]: Convert this to YAML for Hugo front matter.

**Review by adambkaplan — CHANGES_REQUESTED** [review]
I think the YAML bits you overwrote are needed by Hugo. If you look at the deploy preview, the `docs/` path actually renders content, it is just missing a lot of stuff.

**Comment by adambkaplan** [comment]
More info on Hugo front matter: https://gohugo.io/content-management/front-matter/

File: content/en/docs/_index.adoc (hunk 1 above)
Comment (rolfedh) [reply content/en/docs/_index.adoc]: > This is Hugo front matter - I believe Hugo expects this to exist in YAML format somewhere in the source content. Can you try reverting this to see what happens?

I've restored the YAML frontmatter in all the files. The TOC and header all look good now.
However, I'm seeing some other issues with the content that must be fixed before this PR is ready to merge.

**Comment by rolfedh** [comment]
I resolved the "Content Outlines" issues, above, by adding the AsciiDoc TOC directives just below the front matter of each topic:
```
:toc:
:toclevels: 4
```
AND the following parameter values to `config.toml`:
```
  [markup.tableOfContents]
    endLevel = 3
    ordered = false
    startLevel = 2
```

NOTE: The value of `toclevels` must be equal to, or less than, the value of `endLevel`.

**Pull request closed without merging**